source/common.h

```diff
@@ -396,7 +396,7 @@ class cmdline_processor
             }
             print("  -");
             auto n = flag.name.substr(0, flag.unique_prefix);
-            if (flag.unique_prefix < flag.name.length()) {
+            if (flag.unique_prefix < static_cast<ssize_t>(flag.name.length())) {
```

Comment:
How about:
```suggestion
            if (flag.unique_prefix < std::ssize(flag.name)) {
```
Same for the other size/length issues.

Reply:
Would it not be better to change `unique_prefix` to be `size_type`, since only it's being compared with `size_type`, and passed to functions that take `size_type`?